### PR TITLE
Revamped: Resource Request Page

### LINKED ISSUE #2976: Revamp : Resource Request Page
15a

- [ ] The category and subcategory box is having a lot of white space
- [ ] sub category is not fully visible
- [ ] change "is this an emergency to" yes or no option to drop list and if possible change question to a more suitable one
- [ ] the drop list is having curve, remove the curve as it doesn't feel good for the UI

### PR DESCRIPTION
**Pull request by cp-Coder**
# Updates

- [x] Fixes #2976
- [x] Revamped UI of `Resource Request` Page

### CODE REVIEW

File: src/Components/Resource/ResourceCreate.tsx [hunk 1]
```diff
@@ -98,6 +88,9 @@ export default function ResourceCreate(props: resourceProps) {
   const dispatchAction: any = useDispatch();
   const [isLoading, setIsLoading] = useState(false);
   const [facilityName, setFacilityName] = useState("");
+  const [category, setCategory] = useState("OXYGEN");
+  const [subCategory, setSubCategory] = useState("UNSPECIFIED");
+  const [isEmergency, setIsEmergency] = useState("false");
```
Comment (developedBySJ) [inline src/Components/Resource/ResourceCreate.tsx]: Move this state to `state.form`.  And use `useReducer` dispatch

**Comment by aeswibon** [comment]
> @cp-Coder Please add a preview screenshot 🙂

image

File: src/Components/Resource/ResourceCreate.tsx (hunk 1 above)
Comment (aeswibon) [reply src/Components/Resource/ResourceCreate.tsx]: @developedBySJ I am using SelectMenu which requires a state variable to change the selected value. That's why using `state.form` for setting form data will give an error.

Comment (developedBySJ) [reply src/Components/Resource/ResourceCreate.tsx]: You can change the state value using the dispatch function. To be more specific use `handleValueChange` or `handleChange` to update state values.

**Comment by nihal467** [comment]
@cp-Coderimage

**Comment by nihal467** [comment]
@cp-Coder 
image

- [ ] i can scroll down a lot

**Comment by aeswibon** [comment]
@nihal467 I am not facing such issue
image

**Comment by nihal467** [comment]
@cp-Coder test failed,

image

- [ ] make the page responsive
- [ ] make the cancel and submit button in two lines in the responsive view with the same length

image

- [ ] can we have the emergency button on the same line as category and subcategory are there, its just an opinion, before working on this, taking a second opinion also will be good

**Comment by mathew-alex** [comment]
:wave: Hi, @cp-Coder,
Conflicts have been detected against the base branch. Please rebase your branch against the base branch.

***

This message is automatically generated by [prince-chrismc/label-merge-conflicts-action](https://github.com/prince-chrismc/label-merge-conflicts-action) so don't hesitate to report issues/improvements there.

**Comment by aeswibon** [comment]
Fixed in PR #3547

**Pull request closed without merging**